component/src/main/java/org/wso2/extension/siddhi/execution/env/ResourceBatchWindowProcessor.java

```diff
@@ -0,0 +1,378 @@
+                                attributeExpressionExecutors[2]).getValue()));
+                    } else if (attributeExpressionExecutors[2].getReturnType() == Attribute.Type.LONG) {
+                        timeInMilliSeconds = Long.parseLong(String.valueOf(((ConstantExpressionExecutor)
+                                attributeExpressionExecutors[2]).getValue()));
+                    } else {
+                        throw new SiddhiAppValidationException("ResourceBatch window's 3rd parameter " +
+                                "'time.in.milliseconds' should be either be a constant (of type int or long), " +
+                                "but found " + attributeExpressionExecutors[2].getReturnType());
+                    }
+                } else {
+                    throw new SiddhiAppValidationException("ResourceBatch window's 3rd parameter " +
+                            "'time.in.milliseconds' should either be a constant (of type int or long)");
+                }
+            }
+        } else {
+            throw new SiddhiAppValidationException(
+                    "Resource batch window should only have two or three parameters, but found "
+                            + attributeExpressionExecutors.length + " input attributes");
+        }
+    }
+
+    @Override
+    protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processor nextProcessor,
+                           StreamEventCloner streamEventCloner) {
+        synchronized (this) {
+            long currentTime = siddhiAppContext.getTimestampGenerator().currentTime();
+            ComplexEventChunk<StreamEvent> outputStreamEventChunk = new ComplexEventChunk<StreamEvent>(true);
+            while (streamEventChunk.hasNext()) {
+                StreamEvent streamEvent = streamEventChunk.next();
+                StreamEvent clonedStreamEvent = streamEventCloner.copyStreamEvent(streamEvent);
+                if (!streamEvent.getType().equals(ComplexEvent.Type.TIMER)) {
+                    Object groupEventMapKey = groupKeyExpressionExecutor.execute(clonedStreamEvent);
+                    if (groupEventMap.get(groupEventMapKey) != null) {
+                        groupEventMap.get(groupEventMapKey).streamEventList.add(clonedStreamEvent);
+                    } else {
+                        List<StreamEvent> list = new ArrayList<>();
+                        list.add(clonedStreamEvent);
+                        groupEventMap.put(groupEventMapKey, new ResourceStreamEventList(list,
+                                clonedStreamEvent.getTimestamp() + timeInMilliSeconds));
+                        if (nextEmitTime == -1) {
+                            nextEmitTime = currentTime + timeInMilliSeconds;
+                            if (scheduler != null) {
+                                scheduler.notifyAt(nextEmitTime);
+                            }
+                        }
```

Comment:
you are only scheduling once but need to reschedule multiple times right ?
change like 
```
long newNextEmitTime = currentTime + timeInMilliSeconds;
if (nextEmitTime == -1 || newNextEmitTime > nextEmitTime) {
       nextEmitTime = newNextEmitTime;
       if (scheduler != null) {
                scheduler.notifyAt(nextEmitTime);
       }
}
```

Reply:
Fixes Added: 18ef02767fb180107b7938fefd40f4153d6f3807